comp/process/submitter/component.go

```diff
@@ -0,0 +1,44 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package submitter implements a component to submit collected data in the Process Agent to
+// supported Datadog intakes.
+package submitter
+
+import (
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// team: processes
+
+// Payload defines payload to be submitted
+type Payload struct {
```

Comment:
Ideally, it would be great to have in `component.go` only `Component` and `Mock` interfaces and `Module` and `Mock`. Can this struct be defined elsewhere?

Reply:
Thank you, let me think about that, I may be reaching a similar conclusion, we would probably have a common interface package for these.

Reply:
Going to defer this for later when the structure of interfaces is a bit more evolved 🙇 .